ckanext/gdi_userportal/logic/action/post.py

```diff
@@ -0,0 +1,35 @@
+# SPDX-FileCopyrightText: 2024 Stichting Health-RI
+# SPDX-FileContributor: PNED G.I.E.
+#
+# SPDX-License-Identifier: Apache-2.0
+
+from ckan.plugins import toolkit
+from ckanext.gdi_userportal.logic.action.translation_utils import (
+    collect_values_to_translate,
+    get_translations,
+    replace_package,
+    replace_search_facets,
+)
+from typing import Dict
+
+
+@toolkit.side_effect_free
+def enhanced_package_search(context, data_dict=None) -> Dict:
+    if data_dict is None:
```

Comment:
# Ensure data_dict is populated
        if data_dict is None:
            if toolkit.request.json:
                data_dict = toolkit.request.json
            else:
                raise toolkit.ValidationError("data_dict or request body cannot be empty")
                
                Add some logging would be nice